src/common/actions/snaps.js

```diff
@@ -42,6 +42,20 @@ export function fetchUserSnaps(owner) {
   };
 }
 
+function shouldFetchUserSnaps(state) {
+  const snaps = state.snaps;
+
+  return !snaps.isFetching;
+}
+
+export function fetchUserSnapsIfNeeded(owner) {
+  return function(dispatch, getState) {
+    if (shouldFetchUserSnaps(getState())) {
```

Comment:
I'm not sure if it is a good place to do this check (and to access the state in the action creator).

Accessing the state in action is an anti-pattern (http://stackoverflow.com/a/35674575).
Even though this use case seem to fit the "conditional dispatch" exception that allows checking the state in action it still introduces a bit of dependency between action and the store. This action here needs to know where in the store we keep the data related to this particular fetch.

On the other hand every fetch should be 'conditional' - we should never fetch data that we already requested (`isFetching` is true). Right now we were checking it in the component that was requesting the data. But maybe we should aim to refactor all the "fetch" actions to use the state and check if `isFetching` is set?

I guess this comment doesn't provide any useful opinion ;)
I'm not agains this change, but it introduces some new patters we didn't have in the app before.